### PR TITLE
Only allow sign-ups from education.gov.uk emails addresses on qa etc.

### PR DESCRIPTION
**Pull request by stevehook**
## Context

We want to restrict use of test environments (qa, review apps and staging) to DfE uses (those with `education.gov.uk` email addresses).

## Changes proposed in this pull request

- [x] `HostEnvironment.dfe_signups_only?` helper
- [x] Add validation to the `SignUpForm` to prevent non-DfE sign-ups depending on environment.
- [x] System spec
- [x] Redirect to a page warning candidates that they can't sign-up without a DfE email address and link to production environment.

image

## Guidance to review

- Does the validation make sense? Originally it was meant to be displayed as a normal validation but now that we are redirecting to a special 'access forbidden' page it's no longer rendered as a validation error on the sign-up form itself. I've left it this way so that it gets captured by our validation error logging.
- Is the copy for the 'access forbidden' page correct?
- I needed a link to the production environment. I couldn't find an existing one in the codebase so I've added a constant to `HostingEnvironment`. Does this make sense or is there a simpler way? Should it be an env var?

## Link to Trello card

https://trello.com/c/M1IWxq2c/2481-only-allow-digitaleducationgovuk-signups-on-qa-staging-review-apps

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by tvararu — APPROVED** [review]
I think this is good enough, maybe @paulrobertlloyd can suggest further improvements :+1:

**Comment by paulrobertlloyd** [comment]
@tvararu Chatted with @stevehook, going to redirect to a page that gives error message, and allows linking to the production environment (which is not possible with an error summary)

**Comment by paulrobertlloyd** [comment]
@stevehook Can we add a back link to the error page?

**Comment by stevehook** [comment]
> @stevehook Can we add a back link to the error page?

thanks @paulrobertlloyd I missed that, done now

File: app/views/candidate_interface/sign_up/external_sign_up_forbidden.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<% content_for :title, t('page_titles.external_sign_up_forbidden') %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_sign_up_path) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+
+    <h2 class="govuk-heading-xl">
+      <%= t('page_titles.external_sign_up_forbidden') %>
+    </h2>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/sign_up/external_sign_up_forbidden.html.erb:9]:
```suggestion
    <h1 class="govuk-heading-xl">
      <%= t('page_titles.external_sign_up_forbidden') %>
    </h1>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Just one tiny change. Otherwise LGTM!

**Review by paulrobertlloyd — APPROVED** [review]

**Pull request merged**